src/components/QuantityButton.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class QuantityButton extends Component {
+  constructor() {
+    super();
+    this.state = {
+      quantity: 0,
+    };
+    this.addQuantity = this.addQuantity.bind(this);
+    this.subtractQuantity = this.subtractQuantity.bind(this);
+    this.setQuantity = this.setQuantity.bind(this);
+  }
+
+  componentDidMount() {
+    this.setQuantity();
+  }
+
+  setQuantity() {
+    const { product: { quantity } } = this.props;
+    this.setState({
+      quantity,
+    });
+  }
+
+  addQuantity() {
+    this.setState((prevState) => ({
+      quantity: prevState.quantity + 1,
```

Comment:
Que tal desestruturar essa `quantity`? 😉 
```suggestion
    this.setState(({ quantity }) => ({
      quantity: quantity + 1,
```